File: backend/src/services/csvExportService.js
const { escapeCsvField } = require('../lib/sanitize')

const CABECALHO = ['Livro', 'Autor', 'Nota', 'Resenha', 'Data']

function campo(valor) {
  return escapeCsvField(valor)
}

function exportarAvaliacoes(avaliacoes) {
  const linhas = [CABECALHO.join(',')]

  for (const avaliacao of avaliacoes) {
    linhas.push(
      [
        campo(avaliacao.livro.titulo),
        campo(avaliacao.livro.autor),
        avaliacao.nota,
        campo(avaliacao.resenha),
        campo(avaliacao.criadoEm.toISOString()),
      ].join(','),
    )
  }

  return linhas.join('\n') + '\n'
}

module.exports = { exportarAvaliacoes }

File: backend/src/main/java/com/alexandria/service/CsvExportService.java
package com.alexandria.service;

import com.alexandria.model.Avaliacao;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CsvExportService {

    private static final String[] CABECALHO = {"Livro", "Autor", "Nota", "Resenha", "Data"};

    public String exportarAvaliacoes(List<Avaliacao> avaliacoes) {
        StringBuilder csv = new StringBuilder();
        csv.append(String.join(",", CABECALHO)).append("\n");

        for (Avaliacao avaliacao : avaliacoes) {
            csv.append(campo(avaliacao.getLivro().getTitulo())).append(",")
                    .append(campo(avaliacao.getLivro().getAutor())).append(",")
                    .append(avaliacao.getNota()).append(",")
                    .append(campo(avaliacao.getResenha())).append(",")
                    .append(campo(avaliacao.getCriadoEm().toString()))
                    .append("\n");
        }

        return csv.toString();
    }

    private String campo(String valor) {
        if (valor == null) {
            return "";
        }
        return "\"" + valor.replace("\"", "\"\"") + "\"";
    }
}
